cli/command/registry/login.go

```diff
@@ -50,7 +52,18 @@ func runLogin(dockerCli *command.DockerCli, opts loginOptions) error {
 		authServer    = command.ElectAuthServer(ctx, dockerCli)
 	)
 	if opts.serverAddress != "" {
-		serverAddress = opts.serverAddress
+		// when a user supplied login url is the default docker registry save the credentials under the default docker registry domain
+		// without this the login will succeed ,but docker pull/push will fail
+		u, _ := url.Parse(opts.serverAddress)
+		if u.Host == "" {
+			// handle urls without schema for backward compatibility and pass the integration tests
+			u, _ = url.Parse("http://" + opts.serverAddress)
+		}
+		if strings.Contains(authServer, u.Host) {
```

Comment:
This doesn't seem defensive enough. Consider if the user tries to `docker login` to `ocker.io` or `d`. Maybe we should go back to the check from the previous PR.